widgets/Layout/withAuthLayout/WithAithLayout.tsx

```diff
@@ -0,0 +1,24 @@
+import { PropsWithChildren, ReactElement } from 'react';
+
+import { Layout } from '@/widgets/Layout/Layout';
+import { Sidebar } from '@/widgets/sidebar';
+import { NextPage } from 'next';
+
+import s from './WithAuthLayout.module.scss';
+
+export const WithAuthLayout: NextPage<PropsWithChildren> = props => {
+  const { children } = props;
+
+  return (
+    <>
+      <Layout>
+        <Sidebar />
+        <div className={s.WithAuthLayout}>{children}</div>
+      </Layout>
+    </>
+  );
+};
+
+export const getWithAuthLayout = (page: ReactElement) => {
```

Comment:
I think this function can be safely removed:
* it literally adds nothing to the layout, and basically does nothing.
* we still have to assign it or its analogue to the `getLayout` method of each page.
* sometimes we may need to make a "special" wrapper for an existing layout component or give it extra logic. In this case it is also easier and more convenient to use a ready-made component as a basis.

For the first two cases a simple code is enough:
```javascript
SomePage.getLayout = (page: ReactElement) => <Layout>{page}</Layout>
```

This is true for all existing layouts.